includes/reports/overview/list-table/class-sensei-reports-overview-list-table-abstract.php

```diff
@@ -137,32 +137,31 @@ public function generate_report() {
 
 		$this->csv_output = true;
 
-		// Handle orderby.
-		// phpcs:ignore WordPress.Security.NonceVerification.Recommended -- No action, nonce is not required.
-		$orderby = sanitize_key( wp_unslash( $_GET['orderby'] ?? '' ) );
-		if ( empty( $orderby ) || ! array_key_exists( esc_html( $orderby ), $this->get_sortable_columns() ) ) {
-			$orderby = '';
-		}
-
-		// Handle order.
-		// phpcs:ignore WordPress.Security.NonceVerification.Recommended -- No action, nonce is not required.
-		$order = sanitize_key( wp_unslash( $_GET['order'] ?? 'ASC' ) );
-		$order = ( 'ASC' === strtoupper( $order ) ) ? 'ASC' : 'DESC';
-
 		$args = array(
-			'number'  => -1,
-			'offset'  => 0,
-			'orderby' => $orderby,
-			'order'   => $order,
+			'number' => -1,
+			'offset' => 0,
 		);
 
 		// Handle search.
 		// phpcs:ignore WordPress.Security.NonceVerification.Recommended -- No action, nonce is not required.
 		$search = sanitize_text_field( wp_unslash( $_GET['s'] ?? '' ) );
+		// phpcs:ignore WordPress.Security.NonceVerification.Recommended -- No action, nonce is not required.
+		$orderby = sanitize_text_field( wp_unslash( $_GET['orderby'] ?? '' ) );
+		// phpcs:ignore WordPress.Security.NonceVerification.Recommended -- No action, nonce is not required.
+		$order = sanitize_text_field( wp_unslash( $_GET['order'] ?? '' ) );
+
 		if ( ! empty( $search ) ) {
 			$args['search'] = esc_html( $search );
 		}
 
+		if ( ! empty( $orderby ) ) {
+			$args['orderby'] = esc_html( $orderby );
+		}
+
+		if ( ! empty( $order ) ) {
+			$args['order'] = esc_html( $order );
+		}
```

Comment:
Maybe you could use the new `get_order(by)_value` methods here? Would something like this work?

```php
$args = array(
	'number'  => -1,
	'offset'  => 0,
	'orderby' => $this->get_orderby_value(),
	'order'   => $this->get_order_value(),
);
```

Reply:
Hey Miro I refactored this part a bit, but I redid this change many times, so here is the link in [code](https://github.com/Automattic/sensei/pull/4973/files#diff-3b0db8ab59647710de5201ea0742e71848c10e6e1cd15321ef8d65fd3b736f05R142) and not single commit